app/controllers/datasets_controller.rb

```diff
@@ -4,9 +4,16 @@ class DatasetsController < ApplicationController
   include DatasetsHelper
 
   def show
+    begin
+      @dataset = current_dataset
+      raise 'Metadata missing' if @dataset['title'].empty?
+
+    rescue
+      render :template => "errors/not_found", :status => 404
+    end
```

Comment:
I'm guessing from the user perspective that the nature of the error will not be of interest but agree that logging the type of error is required. Currently, the conditions under which information is displayed or not are still to be finalised ... This is a first pass on the back of an initial discussion with Jonathon. 

Are there any plans to set up a logging service like https://logentries.com/ so that we can can keep track of errors?